tests/Controllers/BeatmapsControllerSoloScoresTest.php

```diff
@@ -0,0 +1,233 @@
+            ]),
+            // With preference mods
+            'otherUser' => SoloScore::factory()->withData([
+                'total_score' => 1000,
+                'mods' => static::defaultMods(['PF']),
+            ])->create([
+                'beatmap_id' => static::$beatmap,
+                'preserve' => true,
+                'user_id' => static::$otherUser,
+            ]),
+            'otherUserMods' => SoloScore::factory()->withData([
+                'total_score' => 1000,
+                'mods' => static::defaultMods(['HD', 'PF', 'NC']),
+            ])->create([
+                'beatmap_id' => static::$beatmap,
+                'preserve' => true,
+                'user_id' => static::$otherUser,
+            ]),
+            'otherUserModsExtraNonPreferences' => SoloScore::factory()->withData([
+                'total_score' => 1000,
+                'mods' => static::defaultMods(['DT', 'HD', 'HR']),
+            ])->create([
+                'beatmap_id' => static::$beatmap,
+                'preserve' => true,
+                'user_id' => static::$otherUser,
+            ]),
+            'otherUserModsUnrelated' => SoloScore::factory()->withData([
+                'total_score' => 1000,
+                'mods' => static::defaultMods(['FL']),
+            ])->create([
+                'beatmap_id' => static::$beatmap,
+                'preserve' => true,
+                'user_id' => static::$otherUser,
+            ]),
+            // Same total score but achieved later so it should come up after earlier score
+            'otherUser2Later' => SoloScore::factory()->withData(['total_score' => 1000])->create([
+                'beatmap_id' => static::$beatmap,
+                'preserve' => true,
+                'user_id' => User::factory()->state(['country_acronym' => Country::factory()]),
+            ]),
+            'otherUser3SameCountry' => SoloScore::factory()->withData(['total_score' => 1000])->create([
+                'beatmap_id' => static::$beatmap,
+                'preserve' => true,
+                'user_id' => User::factory()->state(['country_acronym' => $countryAcronym]),
+            ]),
+            // Non-preserved score should be filtered out
+            'nonPreserved' => SoloScore::factory()->create([
+                'beatmap_id' => static::$beatmap,
+                'preserve' => false,
+                'user_id' => User::factory()->state(['country_acronym' => Country::factory()]),
+            ]),
+            // Unrelated score
+            'unrelated' => SoloScore::factory()->create([
+                'user_id' => User::factory()->state(['country_acronym' => Country::factory()]),
+            ]),
+        ];
+
+        UserRelation::create([
+            'friend' => true,
+            'user_id' => static::$user->getKey(),
+            'zebra_id' => static::$scores['friend']->user_id,
+        ]);
+
+        Artisan::call('es:index-scores:queue', [
```

Comment:
it does give stack trace in case of coding failure at least

```
1) Tests\Controllers\BeatmapsControllerSoloScoresTest::testQuery
Error: Call to undefined function App\Console\Commands\asb() in /home/edho/osu/osu-web/app/Console/Commands/EsIndexScoresQueue.php:116
Stack trace:
#0 /home/edho/osu/osu-web/app/Console/Commands/EsIndexScoresQueue.php(83): App\Console\Commands\EsIndexScoresQueue->queueIds(Array)
...
```